pinot-broker/src/test/java/org/apache/pinot/broker/broker/BrokerManagedAsyncExecutorProviderTest.java

```diff
@@ -0,0 +1,120 @@
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+public class BrokerManagedAsyncExecutorProviderTest {
+
+    public static BrokerMetrics _brokerMetrics;
+
+    @BeforeClass
+    public void setUp() {
+        _brokerMetrics = new BrokerMetrics(
+                CommonConstants.Broker.DEFAULT_METRICS_NAME_PREFIX,
+                PinotMetricUtils.getPinotMetricsRegistry(null),
+                CommonConstants.Broker.DEFAULT_ENABLE_TABLE_LEVEL_METRICS,
+                Collections.emptyList());
+    }
+
+    @Test
+    public void testExecutorService() throws InterruptedException, ExecutionException {
+        // create a new instance of the executor provider
+        BrokerManagedAsyncExecutorProvider provider = new BrokerManagedAsyncExecutorProvider(_brokerMetrics, 2, 2, 2);
+
+        // get the executor service
+        ThreadPoolExecutor executor = (ThreadPoolExecutor) provider.getExecutorService();
+
+        // submit a task to the executor service and wait for it to complete
+        Future<Integer> futureResult = executor.submit(() -> 1 + 1);
+        Integer result = futureResult.get();
+
+        // verify that the task was executed and returned the expected result
+        assertNotNull(result);
+        assertEquals((int) result, 2);
+
+        // wait for the executor service to shutdown
+        executor.shutdown();
+        executor.awaitTermination(1, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void testGet() {
+        BrokerManagedAsyncExecutorProvider provider = new BrokerManagedAsyncExecutorProvider(_brokerMetrics, 1, 1, 1);
+        ExecutorService executorService = provider.getExecutorService();
+
+        // verify that the executor has the expected properties
+        assertNotNull(executorService);
+        assertTrue(executorService instanceof ThreadPoolExecutor);
+
+        ThreadPoolExecutor threadPoolExecutor = (ThreadPoolExecutor) executorService;
+
+        assertEquals(1, threadPoolExecutor.getCorePoolSize());
+        assertEquals(1, threadPoolExecutor.getMaximumPoolSize());
+
+        BlockingQueue<Runnable> blockingQueue = threadPoolExecutor.getQueue();
+        assertNotNull(blockingQueue);
+        assertTrue(blockingQueue instanceof ArrayBlockingQueue);
+        assertEquals(0, blockingQueue.size());
+        assertEquals(1, blockingQueue.remainingCapacity());
+
+        RejectedExecutionHandler rejectedExecutionHandler = threadPoolExecutor.getRejectedExecutionHandler();
+        assertNotNull(rejectedExecutionHandler);
+        assertTrue(rejectedExecutionHandler
+                instanceof BrokerManagedAsyncExecutorProvider.BrokerThreadPoolRejectExecutionHandler);
+
+        // test that the executor actually executes tasks
+        AtomicInteger counter = new AtomicInteger();
+        for (int i = 0; i < 1; i++) {
+            threadPoolExecutor.execute(() -> counter.incrementAndGet());
+        }
+        assertEquals(counter.get(), 0);
```

Comment:
I think you need to do `assertEquals(counter.get(), 1)`.

Either ways, there's a race condition here which can introduce flakiness in tests. You can use a `CountDownLatch` instead.

Reply:
Thanks, this is fixed.